script.module.cache/lib/cache.py

```diff
@@ -0,0 +1,311 @@
+# -*- coding: utf-8 -*-
+
+"""HTTP File cache and helpers"""
+
+__author__ = "fraser"
+
+import json
+import logging
+import sqlite3
+import time
+from datetime import datetime, timedelta, tzinfo
+from os import path
+
+import xbmc
+import xbmcaddon
+import xbmcvfs
+
+try:
+    import _pickle as cpickle
+except ImportError:
+    import cPickle as cpickle
+
+logger = logging.getLogger(__name__)
+
+HTTP_DATE_FORMAT = "%a, %d %b %Y %H:%M:%S %Z"
+ADDON = xbmcaddon.Addon()
+ADDON_PROFILE = ADDON.getAddonInfo("profile")
```

Comment:
Should be decoded into UTF-8 in Python 2

Reply:
Done